wolfcrypt/src/port/maxim/maxq10xx.c

```diff
@@ -0,0 +1,3376 @@
+        totalkeylen = keycomplen;
+
+    }
+    else {
+        WOLFSSL_MSG("MAXQ: key algo not supported");
+        return NOT_COMPILED_IN;
+    }
+
+    tls13_server_key_algo = p_cert->keyOID;
+    tls13_server_key_len = keycomplen;
+    sign_algo = mxq_get_sign_alg_from_sig_oid(p_cert->signatureOID);
+
+    if (sign_algo == ALGO_INVALID) {
+        WOLFSSL_MSG("MAXQ: signature algo not supported");
+        return NOT_COMPILED_IN;
+    }
+#endif
+
+    rc = wolfSSL_CryptHwMutexLock();
+    if (rc != 0) {
+        return rc;
+    }
+
+    mxq_rc = MXQ_Build_EC_Cert(certdata, &certdatalen, key_type, keyparam,
+                               keycomplen, totalkeylen, pk_offset,
+                               p_cert->certBegin,
+                               (p_cert->sigIndex - p_cert->certBegin),
+                               p_cert->maxIdx, sign_algo, ROOT_CA_CERT_OBJ_ID,
+                               MXQ_KEYUSE_VERIFY_KEY_CERT, ALGO_ECDSA_SHA_any,
+                               MXQ_KEYUSE_DATASIGNATURE, ALGO_ECDSA_SHA_any,
+                               (mxq_u1 *)p_cert->source);
+    if (mxq_rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: MXQ_Build_EC_Cert() failed");
+        wolfSSL_CryptHwMutexUnLock();
+        return WC_HW_E;
+    }
+
+#if defined(WOLFSSL_MAXQ108X)
+    if (tls13_server_cert_id == -1) {
+        tls13_server_cert_id = alloc_temp_key_id();
+        if (tls13_server_cert_id == -1) {
+            WOLFSSL_ERROR_MSG("MAXQ: alloc_temp_key_id() failed");
+            wolfSSL_CryptHwMutexUnLock();
+            return WC_HW_E;
+        }
+    }
+
+    /* temporary certificate stored in object id cert_id */
+    mxq_rc = MXQ_ImportChildCert(tls13_server_cert_id, certdata, certdatalen);
+#else
+    /* temporary certificate stored in object id 0 */
+    mxq_rc = MXQ_ImportChildCert(0, certdata, certdatalen);
+#endif
+
+    if (mxq_rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: MXQ_ImportChildCert() failed");
+        wolfSSL_CryptHwMutexUnLock();
+        return WC_HW_E;
+    }
+
+    wolfSSL_CryptHwMutexUnLock();
+    return 0;
+}
+
+static int maxq10xx_process_server_sig_kex(WOLFSSL* ssl, byte p_sig_algo,
+               const byte* p_sig, word32 p_sig_len,
+               const byte* p_rand, word32 p_rand_len,
+               const byte* p_server_params, word32 p_server_params_len)
+{
+    int rc;
+    mxq_err_t mxq_rc;
+    if (ssl->specs.kea != ecc_diffie_hellman_kea) {
+        WOLFSSL_MSG("MAXQ: key exchange algo not supported");
+        return NOT_COMPILED_IN;
+    }
+
+    if (ssl->ecdhCurveOID != ECC_SECP256R1_OID) {
+        WOLFSSL_MSG("MAXQ: key curve not supported");
+        return NOT_COMPILED_IN;
+    }
+
+    if (p_sig_algo != ecc_dsa_sa_algo) {
+        WOLFSSL_MSG("MAXQ: signature algo not supported");
+        return NOT_COMPILED_IN;
+    }
+
+    rc = wolfSSL_CryptHwMutexLock();
+    if (rc != 0) {
+        return rc;
+    }
+
+    /* NOTE: this funciton also does verification of the signature as well! */
```

Comment:
Please spell check this file. `funciton`?

Reply:
Spell check run.  Stay tuned for next push.